src/site/content/en/blog/ecommerce-notifications-best-practice/index.md

```diff
@@ -0,0 +1,360 @@
+
+However, customers might be less willing to tolerate this behavior from sites
+they are not invested in. To build trust and retain customers, implement
+notifications in a way that respects the customer and adds to their experience.
+
+{% Aside %}
+
+While the current web platform lets you request notification permissions at
+any point in the user journey, this could change. Implement notifications
+respectfully to future-proof your app!
+
+{% endAside %}
+
+#### When to prompt a user to permit push notifications
+
+A permission request creates a popup on your customer’s screen that they must
+deal with before proceeding. Make sure you don’t interrupt customers mid-way
+through performing a task.
+
+Don’t ask for permission:
+
+*   While a customer is making a purchase.
+
+    A popup is likely to jolt someone away from a multi-step task like
+    completing a payment.
+
+*   While a customer is actively viewing content.
+
+    A customer who is actively viewing content–for example, scrolling through a
+    list of search results, or browsing a category or product page–probably has
+    a goal in mind. If a popup appears at the wrong moment, the customer might
+    abandon their goal or leave the site altogether.
+
+{% Aside %}
+
+Since mobile devices are such an intimate part of our lives, we are uniquely
+connected to them. In particular, mobile users are more likely than desktop
+users to be acting on their emotions. You don’t want to interrupt someone who
+is making a decision in-the-moment!
+
+{% endAside %}
+
+Do ask for permission:
+
+*   When the value of notifications is obvious.
+
+    For example, an ideal time to prompt the customer for permission to send
+    push notifications is immediately after they have completed a purchase and
+    might want to receive updates about shipping.
+
+*   When the customer has completed a user journey that has clear followup
+    potential.
+
+    Another good time to ask is when a customer has finished doing something
+    that precedes a possible later action. For example, a customer who has
+    searched for and clicked on a product that is out of stock might want
+    to be notified when you get more in.
+
+*   In response to a customer action. 
+
+    Using your app’s own UI, you can make sure that the push notification
+    request popup appears when the customer intends to subscribe. This avoids
+    situations in which customers block your app from sending notifications.
+
+These techniques also let you maintain consistent site branding, voice, and
+tone.
+
+#### Example implementation
+
+Our sample app lets users summon the push notifications popup by indicating
+their interest in a product: 
+
+```html
+<h2>{{product.name}}</h2>
+<p class="price">${{product.price}}</p>
+<form action="/{{product.url}}/cart" method="post">
+  <button class="add-to-cart">Add to Cart</button>
+  <button class="notify">Notify me about this item</button>
```

Comment:
Maybe highlight this line? It wasn't immediately obvious to me what this code snippet was for.